classes/tool/tool_base.php

```diff
@@ -144,4 +146,13 @@ public function trackurl($urlencode = false, $leadingslash = false): string {
         return $trackurl;
     }
 
+    /**
+     * Override in child to implement.
```

Comment:
Can you please flesh this out more with how this should work generically, eg if someone wanted to come and add support for GA later on

Reply:
That should be more descriptive now.